mirrord/config/derive/src/config/field.rs

```diff
@@ -137,21 +137,18 @@ impl ConfigField {
             ..
         } = &self;
 
+        // Rest of flow is irrelevant for nested config.
+        if flags.nested {
+            return quote! { #ident: self.#ident.unwrap_or_default().generate_config()? };
```

Comment:
Breaking here for nested types will prevent us from using `.layer` on nested types in the future (if we want to keep the layer design for augmenting config or just messaging the user with the configs)

Reply:
how can a nested type need flag though? it's probably its children that will need it anyway?

Reply:
I was thinking if we want some tracing layer or something similar

Reply:
Not important just nitpick